### PR TITLE
Add support for Bloom filters on unsigned integer columns in Parquet tables

### LINKED ISSUE #9769: Add support for Bloom filters on unsigned integer columns in Parquet tables
### Is your feature request related to a problem or challenge?

_No response_

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by progval**
## Which issue does this PR close?

Closes #9769.

## What changes are included in this PR?

Also fixed `prune_int $bits _scalar_fun_and_eq` (on signed integers) to run the query it should have been running.

## Are these changes tested?

They are not, but the code is straightforward so I don't think tests would be useful.

## Are there any user-facing changes?

It should be invisible other than a performance improvement.

### CODE REVIEW

**Review by alamb — COMMENTED** [review]
Thank you very much @progval  🙏 

Can we please add tests for this code (so that we don't accidentally break it in some future refactoring?)

**Comment by alamb** [comment]
> Thank you very much @progval 🙏
> 
> Can we please add tests for this code (so that we don't accidentally break it in some future refactoring?)

Hi @progval  -- do you think you will have time to add some tests briefly

I noticed that https://github.com/apache/arrow-datafusion/pull/9778 only adds tests for the signed (`Int8`/`Int16`/etc variants). I'll go get that PR ready to merge per our discussion as maybe you wrote it to add tests for this one 🤔

**Comment by progval** [comment]
> as maybe you wrote it to add tests for this one 🤔

yes that's what I had in mind

**Comment by alamb** [comment]
Ok, https://github.com/apache/arrow-datafusion/pull/9778 has now been merged. Sorry for the delay

**Comment by progval** [comment]
Tests added. There is some code in common with https://github.com/apache/arrow-datafusion/pull/9888 but they shouldn't merge-conflict

**Comment by progval** [comment]
> but they shouldn't merge-conflict

They did, but I just fixed it.

And now it seems tests are failing because I merged the latest on `main`, which is also failing.

**Review by alamb — APPROVED** [review]
Thank you @progval  -- this looks great to me

The test failures appear to be due to https://github.com/apache/arrow-datafusion/issues/9891

I took the liberty of merging up to main to get the CI to pass

File: datafusion/core/tests/parquet/row_group_pruning.rs [hunk 1]
```diff
@@ -452,6 +452,144 @@ int_tests!(16, correct_bloom_filters: false);
 int_tests!(32, correct_bloom_filters: true);
 int_tests!(64, correct_bloom_filters: true);
 
+// $bits: number of bits of the integer to test (8, 16, 32, 64)
```
Comment (alamb) [inline datafusion/core/tests/parquet/row_group_pruning.rs:455]: I didn't realize `UInt8` and `UInt16` were also affected. 💯  for testing

**Pull request merged**